opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/TracerAutoConfigurationWithWrapperTest.java

```diff
@@ -0,0 +1,119 @@
+package io.opentracing.contrib.spring.web.autoconfig;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.Field;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.NoopSpan;
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.util.GlobalTracer;
+
+@SpringBootTest(
+        classes = {TracerAutoConfigurationWithWrapperTest.SpringConfiguration.class,
+                TracerAutoConfigurationWithWrapperTest.TestTracerBeanPostProcessor.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+public class TracerAutoConfigurationWithWrapperTest {
+
+    // Temporary solution until https://github.com/opentracing/opentracing-java/issues/170 resolved
+    private static void _setGlobal(Tracer tracer) {
+        try {
+            Field globalTracerField = GlobalTracer.class.getDeclaredField("tracer");
+            globalTracerField.setAccessible(true);
+            globalTracerField.set(null, tracer);
+            globalTracerField.setAccessible(false);
+        } catch (Exception e) {
+            throw new RuntimeException("Error reflecting globalTracer: " + e.getMessage(), e);
+        }
+    }
+
+    @BeforeClass
+    public static void clearGlobalTracer() {
+        _setGlobal(NoopTracerFactory.create());
+    }
+
+    @Autowired
+    private Tracer tracer;
+
+    @Configuration
+    @EnableAutoConfiguration
+    public static class SpringConfiguration {
+    }
+
+    @Test
+    public void testGetAutoWiredTracer() {
+        assertNotNull(tracer);
```

Comment:
Can we directly test an expected type?

Reply:
Don't think so - GlobalTracer doesn't provide access to the actual tracer. That is why I'm testing the type of the span a couple of lines down, to double check it is using the NoopTracer.

Reply:
The bean is not `GlobalTracer` but `TracerWrapper`

Reply:
Ah yes - it used to be the `GlobalTracer` - ok will update.